lib/active_record/virtual_attributes.rb

```diff
@@ -85,16 +85,13 @@ def virtual_attribute_names
         end
       end
 
-      private
-
-      def load_schema!
-        super
-
-        virtual_attributes_to_define.each do |name, (type, options)|
-          type = type.call if type.respond_to?(:call)
-          type = ActiveRecord::Type.lookup(type, **options.except(:uses, :arel)) if type.kind_of?(Symbol)
-
-          define_virtual_attribute(name, type)
+      def attribute_types
+        @attribute_types || super.tap do | hash|
+          virtual_attributes_to_define.each do |name, (type, options)|
+            type = type.call if type.respond_to?(:call)
+            type = ActiveRecord::Type.lookup(type, **options.except(:uses, :arel)) if type.kind_of?(Symbol)
+            hash[name] = type
+          end
```

Comment:
wow, this is the only change for 7.2 after those others?  Crazy!

Reply:
So #186 is all the changes. So I wanted to run cross repo on that and make those changes